activerecord/test/cases/persistence_test.rb

```diff
@@ -118,6 +118,15 @@ def test_increment_attribute_by
     assert_equal 59, accounts(:signals37, :reload).credit_limit
   end
 
+  def test_increment_concurency
+    a1 = accounts(:signals37)
+    initial_credit = a1.credit_limit
+    a2 = Account.find(accounts(:signals37).id)
+    a1.increment!(:credit_limit)
+    a2.increment!(:credit_limit)
+    assert_equal initial_credit + 2, a1.reload.credit_limit
```

Comment:
This patch doesn't fixes that. We can not make everything perfect from one patch. I would prefer to apply it afterwards.